lexibank_bds.py

```diff
@@ -0,0 +1,138 @@
+    )
+
+
+@attr.s
+class Doculect(Language):
+    Dataset = attr.ib(default=None)
+    ChineseName = attr.ib(default=None)
+    Family = attr.ib(default=None)
+    SubGroup = attr.ib(default=None)
+    Datasets = attr.ib(default=None)
+
+
+class Dataset(BaseDataset):
+    dir = pathlib.Path(__file__).parent
+    id = "borrowing-detection-study"
+    language_class = Doculect
+    lexeme_class = Word
+
+    _wlname = 'seabor.sqlite3'
+
+    def wl(self):
+        return Wordlist(str(self.raw_dir / self._wlname))
+
+    def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
+        return super().cldf_specs()
+
+    def cmd_download(self, args):
+        fetch("seabor", outdir=self.raw_dir, remote_dbase=self._wlname, base_url="https://digling.org/edictor/")
+        wl = self.wl()
+        datasets = defaultdict(lambda: defaultdict(list))
+        langs = defaultdict(lambda: defaultdict(list))
+        for idx in wl:
+            datasets[wl[idx, 'dataset']][wl[idx, 'concept']] += [wl[idx, 'doculect']]
+            langs[wl[idx, 'dataset']][wl[idx, 'doculect']] += [wl[idx, 'concept']]
+
+        tab = []
+        for dataset in datasets:
+            tab += [[dataset, len(datasets[dataset]), len(langs[dataset])]]
+        print(tabulate(tab))
+
+    def cmd_makecldf(self, args):
+        from lingrex.cognates import common_morpheme_cognates
+        from lingrex.borrowing import internal_cognates, external_cognates
+
+        #wl = self.wl()
+        #internal_cognates(wl, runs=10000, ref="autocogids", partial=True,
+        #          method="lexstat", threshold=0.55, cluster_method="infomap")
+        #common_morpheme_cognates(wl, ref="autocogid", cognates="autocogids",
+        #                 morphemes="automorphemes")
+        #external_cognates(wl, cognates="autocogid", ref="autoborid")
+
+        wl = Wordlist('wordlist-borrowings.tsv')
+        args.writer.add_languages()
+
+        concepts = set()
+        header = [k for k, _ in sorted(wl.header.items(), key=lambda i: i[1])]
+        for index in wl:
+            row = dict(zip(header, wl[index]))
+            if row['concept'] not in concepts:
+                args.writer.add_concept(
+                    ID=slug(row['concept']),
+                    Concepticon_Gloss=row['concept'].upper(),
+                    Name=row['concept_in_source'])
+                concepts.add(row['concept'])
+            args.writer.add_form_with_segments(
+                ID='{}-{}'.format(row['dataset'], row['lexibank_id']),
+                Language_ID=row['doculect'],
+                Parameter_ID=slug(row['concept']),
+                Value=row['value'],
+                Form=row['form'],
+                Segments=row['tokens'],
+                autoborid=row['autoborid'],
+                autocogid=row['autocogid'],
+                autocogids=row['autocogids'],
+            )
+
+        # autocogids, autocogid, autoborid.
```

Comment:
lots of the actual values can be discarded, maybe the lexibank_id is useful and the alignmetns (but one would have to rerun them, they depend on the cognate set chosen as well). The "uborid" is my user assessment, so this may be useful to have there as well.